lib/src/Uncertainty/Distribution/JointDistribution.cxx

```diff
@@ -308,6 +323,32 @@ JointDistribution::DistributionCollection JointDistribution::getDistributionColl
 }
 
 
+/* Kernel accessor */
+void JointDistribution::setKernel(const Distribution & kernel)
+{
+  // We check if the kernel has a dimension compatible with the one of the object,
+  // especially if the object has already been created and has a collection of distribution
+  if (getDimension() != 0)
+  {
+    if (getDimension() != kernel.getDimension())
+      throw InvalidArgumentException(HERE) << "Kernel has a dimension different from the JointDistribution's";
+  }
+  else
+    setDimension(kernel.getDimension());
+
+  kernel_ = kernel;
+  isAlreadyComputedCovariance_ = false;
+  // We ensure that the kernel has the same description than the JointDistribution
+  kernel_.setDescription(getDescription());
+}
```

Comment:
Couldn't you check that the support of the "kernel" (see my post above for other name ideas) is the unit cube?